### PR TITLE
fix symlink nested package

### PR DESCRIPTION
**Pull request by yihuang**
- fix a symlink bug when there's nested packages
- fix a `relPath` missing bug when generating gomod2nix.toml.
- added test cases for them.

### CODE REVIEW

File: builder/default.nix [hunk 1]
```diff
@@ -71,8 +71,6 @@ let
             go run ${./symlink.go}
             ${lib.concatStringsSep "\n" localReplaceCommands}
 
-            find vendor
```
Comment (yihuang) [inline builder/default.nix:74]: I think it output too much noice.

File: tests/ethermint/go.mod [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+go 1.17
+
+require (
+	github.com/btcsuite/btcd v0.22.1
+	github.com/btcsuite/btcutil v1.0.3-0.20201208143702-a53e38424cce
+)
+
+require (
+	github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1 // indirect
```
Comment (yihuang) [inline tests/ethermint/go.mod:11]: the nested package that trigger the bug is this one.

File: builder/default.nix (hunk 1 above)
Comment (adisbladis) [reply builder/default.nix:74]: I agree, it was a temporary debugging thing that was left there by accident.

File: go.mod [hunk 3]
```diff
@@ -7,5 +7,6 @@ require (
 	github.com/orivej/go-nix v0.0.0-20180830055821-dae45d921a44
 	github.com/sirupsen/logrus v1.7.0
 	golang.org/x/mod v0.3.0
+	golang.org/x/sys v0.0.0-20220519141025-dcacdad47464 // indirect
```
Comment (yihuang) [inline go.mod:10]: this is necessary to build under go 1.18.

File: tests/ethermint/go.mod [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	github.com/golang/protobuf v1.5.2 // indirect
+	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9 // indirect
+	golang.org/x/net v0.0.0-20201021035429-f5854403a974 // indirect
+	golang.org/x/sys v0.0.0-20210119212857-b64e53b001e4 // indirect
+	golang.org/x/text v0.3.3 // indirect
+	google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013 // indirect
+	google.golang.org/protobuf v1.27.1 // indirect
+)
+
+replace github.com/btcsuite/btcd/chaincfg/chainhash => github.com/yihuang/btcd/chaincfg/chainhash v1.0.2-0.20220517215929-cee92e09adcf
```
Comment (yihuang) [inline tests/ethermint/go.mod:23]: this replace trigger the missing `relPath` bug.

File: builder/symlink.go [hunk 5]
```diff
@@ -76,7 +79,21 @@ func main() {
...
+					if err := os.Symlink(innerSrc, dst); err != nil {
+						// assume it's an existing directory, try to link the directory content instead.
+						// TODO should we do this recursively
+						files, err := ioutil.ReadDir(innerSrc)
+						if err != nil {
+							panic(err)
+						}
+						for _, f := range files {
+							if err := os.Symlink(filepath.Join(innerSrc, f.Name()), filepath.Join(dst, f.Name())); err != nil {
+								fmt.Println("ignore symlink error", filepath.Join(innerSrc, f.Name()), filepath.Join(dst, f.Name()))
```
Comment (yihuang) [inline builder/symlink.go:93]:
```suggestion
```
this log is noisy too, not sure if we should just ignore the error.

**Pull request merged**